apps/dash-ocean-optics/app.py

```diff
@@ -0,0 +1,491 @@
+strobe_period = Control(
+    "continuous-strobe-period",
+    "Strobe pd. (μs)",
+    "NumericInput",
+    {
+        "id": "continuous-strobe-period-input",
+        "className": "control__dropdowns",
+        "max": 100,
+        "min": 1,
+        "size": 150,
+        "value": 1,
+    },
+)
+controls.append(strobe_period)
+
+# light sources
+light_sources = Control(
+    "light-source",
+    "Light source",
+    "Dropdown",
+    {
+        "id": "light-source-input",
+        "className": "light-source-dropdown",
+        "options": spec.light_sources(),
+        "placeholder": "select light source",
+        "value": "l2" if DEMO else "",
+    },
+)
+controls.append(light_sources)
+
+base_intro = """This app was created to act as an interface for an Ocean Optics \
+            spectrometer. Use controlling elements to control various \
+            properties of the instrument; the integration time, the number of \
+            scans to average over, the strobe and strobe period, and the \
+            light source.
+"""
+
+extend_intro = """This app was created to act as an interface for an Ocean Optics 
+spectrometer. Use controlling elements to control various 
+properties of the instrument; the integration time, the number of 
+scans to average over, the strobe and strobe period, and the
+light source.
+
+Clicking "Update" after putting in the desired settings will 
+result in your parameter settings being sent to the device. A status message 
+will appear below the button indicating which commands, if any, 
+were unsuccessful; below the unsuccessful commands, a list of
+successful commands can be found.
+           
+The dial labelled "Light intensity" will affect the current 
+selected light source, if any. The switch labelled autoscale 
+plot will change the axis limits of the plot to fit all of the 
+data. Please note that the animations and speed of the graph will 
+improve if this autoscale is turned off, and that it will not be 
+possible to zoom in on any portion of the plot if it is turned 
+on.
+"""
+
+page_layout = [
+    html.Div(
+        [
+            html.Div(
+                [
+                    html.Img(
+                        src=app.get_asset_url("dash-daq-logo.png"), className="logo"
+                    ),
+                    html.Div(
+                        id="controls",
+                        title="All of the spectrometer parameters that can be changed.",
+                        children=[ctrl.create_ctrl_div(True) for ctrl in controls],
+                    ),
+                    html.Div(html.Label("Light Intensity"), className="control"),
+                    html.Div(
+                        [
+                            html.Div(
+                                [
+                                    daq.Knob(
+                                        id="light-intensity-knob",
+                                        size=110,
+                                        color="#565656",
```

Comment:
probably being incredibly pedantic but color could maybe go into app.css instead

Reply:
hmm I'd leave it here since color is a property name for daq.Knob() instead of a inline-style.